### DISCUSSION TITLE
permission problem with Azure Blob Storage

### DISCUSSION THREAD

**Discussion by JanneHarju** [drakkan/sftpgo#547] [Q&A] [answered]
Hello again,

Now I managed to get user configured with UI and managed to connected with FileZilla. Now I have problem with permissions. I assume that my permisson problem is related to Azure permissions. I followed this guide https://github.com/drakkan/sftpgo/blob/main/docs/azure-blob-storage.md
and used 1. option with account key. 

Now when I trying to upload file with FileZilla, sftpgo is trying to create folder and says this
`"message":"root directory \"/project_demo\" for user \"projectadmin\" does not exist, try to create, mkdir error: mkdir /project_demo: permission denied"`

How can I set correct permissions to Azure? I have tried add Stored access policies for container. Should I use Access Control(IAM) some how?

**Comment by drakkan** [comment, marked as answer]
Hi,

this isn't an Azure issue. Please leave the user home dir empty or set it to a path where the user running docker (id 1000) can write

**Reply by JanneHarju** [reply]
That fixed it. Thank you. I must ask what is difference and use cases for configuring folder to azure blob storage and configuring user to use azure blob storage?

**Reply by drakkan** [reply]
If you are asking why you also need a local writable folder this is because an SFTP client upload/download using random access reads/writes, so we keep the temporay file on local disk until a download/upload completes. This means that if you want to upload a 10GB file you also need 10GB of local disk space. SFTPGo uses unliked files, so partial files cannot remain on local disk even if the process is killed while a download/upload is in progress.

If that doesn't answer your question, please be more specific, thank you

**Reply by drakkan** [reply]
You are asking about virtual folders, sorry for misunderstanding, please take a look [here](https://github.com/drakkan/sftpgo/blob/main/docs/virtual-folders.md) for details.

Even if you only have one azure bucket, you could assign to each user a specific virtual directory (setting a key prefix), so for example each user writes to `users/username` and you could share a bucket portion among different users defining a folder that can write to `folders/foldername`. It depends on your use case, vIrtual folders could also be useless for you

**Reply by JanneHarju** [reply]
Yes. That is what I asked. Thank you. Now it starts to be quite clear.
